### PR TITLE
Don't Agree flags

### PR DESCRIPTION
**Pull request by davidgljay**
## What does this PR do?

Moves "Don't Agree" flags to their own datatype and removes them from the moderation queue.

## How do I test this PR?

1. Report a comment.
2. Select "don't agree" as a reason.
3. Flag on the report button should turn red.
4. Reload.
5. Flag on the report button should still be red.
6. Click the admin panel.
7. The "don't agree" flag should not appear the flagged comments section.

## Note
There's a lot of boilerplate required to add a new action type at this point, we should think about creative ways to minimize it for plugin developers.

### CODE REVIEW

File: client/coral-plugin-flags/FlagBio.js [hunk 1]
```diff
@@ -1,35 +0,0 @@
-import React from 'react';
```
Comment (davidgljay) [inline client/coral-plugin-flags/FlagBio.js:1]: This file is no longer used, so I went ahead and blew it away.

**Comment by gabelula** [comment]
There are some errors in the console when I click the moderation queue in the admin:

```
Error in observer.error 
invariant@http://localhost:3000/client/coral-admin/bundle.js:175:15
traverseAllChildrenImpl@http://localhost:3000/client/coral-admin/bundle.js:16328:86
traverseAllChildren@http://localhost:3000/client/coral-admin/bundle.js:16356:10
ReactChildReconciler.instantiateChildren@http://localhost:3000/client/coral-admin/bundle.js:33894:7
ReactMultiChild.Mixin._reconcilerInstantiateChildren@http://localhost:3000/client/coral-admin/bundle.js:38800:20
ReactMultiChild.Mixin.mountChildren@http://localhost:3000/client/coral-admin/bundle.js:38839:22
ReactDOMComponent.Mixin._createInitialChildren@http://localhost:3000/client/coral-admin/bundle.js:35835:27
ReactDOMComponent.Mixin.mountComponent@http://localhost:3000/client/coral-admin/bundle.js:35660:7
ReactReconciler.mountComponent@http://localhost:3000/client/coral-admin/bundle.js:4744:18
ReactMultiChild.Mixin.mountChildren@http://localhost:3000/client/coral-admin/bundle.js:38851:28
ReactDOMComponent.Mixin._createInitialChildren@http://localhost:3000/client/coral-admin/bundle.js:35835:27
ReactDOMComponent.Mixin.mountComponent@http://localhost:3000/client/coral-admin/bundle.js:35660:7
ReactReconciler.mountComponent@http://localhost:3000/client/coral-admin/bundle.js:4744:18
ReactCompositeComponentMixin._updateRenderedComponent@http://localhost:3000/client/coral-admin/bundle.js:34842:24
ReactCompositeComponentMixin._performComponentUpdate@http://localhost:3000/client/coral-admin/bundle.js:34801:5
ReactCompositeComponentMixin.updateComponent@http://localhost:3000/client/coral-admin/bundle.js:34722:7
ReactCompositeComponentMixin.receiveComponent@http://localhost:3000/client/coral-admin/bundle.js:34624:5
ReactReconciler.receiveComponent@http://localhost:3000/client/coral-admin/bundle.js:4823:5
ReactCompositeComponentMixin._updateRenderedComponent@http://localhost:3000/client/coral-admin/bundle.js:34831:7
ReactCompositeComponentMixin._performComponentUpdate@http://localhost:3000/client/coral-admin/bundle.js:34801:5
ReactCompositeComponentMixin.updateComponent@http://localhost:3000/client/coral-admin/bundle.js:34722:7
ReactCompositeComponentMixin.performUpdateIfNecessary@http://localhost:3000/client/coral-admin/bundle.js:34638:7
ReactReconciler.performUpdateIfNecessary@http://localhost:3000/client/coral-admin/bundle.js:4855:5
runBatchedUpdates@http://localhost:3000/client/coral-admin/bundle.js:2628:5
Mixin.perform@http://localhost:3000/client/coral-admin/bundle.js:6180:13
Mixin.perform@http://localhost:3000/client/coral-admin/bundle.js:6180:13
.perform@http://localhost:3000/client/coral-admin/bundle.js:2567:12
flushBatchedUpdates@http://localhost:3000/client/coral-admin/bundle.js:2650:7
Mixin.closeAll@http://localhost:3000/client/coral-admin/bundle.js:6246:11
Mixin.perform@http://localhost:3000/client/coral-admin/bundle.js:6193:11
ReactDefaultBatchingStrategy.batchedUpdates@http://localhost:3000/client/coral-admin/bundle.js:38125:7
enqueueUpdate@http://localhost:3000/client/coral-admin/bundle.js:2678:5
enqueueUpdate@http://localhost:3000/client/coral-admin/bundle.js:15610:3
ReactUpdateQueue.enqueueSetState@http://localhost:3000/client/coral-admin/bundle.js:15795:5
ReactComponent.prototype.setState@http://localhost:3000/client/coral-admin/bundle.js:15238:3
graphql/wrapWithApolloComponent/GraphQL</GraphQL.prototype.forceRenderChildren@http://localhost:3000/client/coral-admin/bundle.js:53949:21
graphql/wrapWithApolloComponent/GraphQL</GraphQL.prototype.subscribeToQuery/next@http://localhost:3000/client/coral-admin/bundle.js:53926:21
graphql/wrapWithApolloComponent/GraphQL</GraphQL.prototype.subscribeToQuery/handleError@http://localhost:3000/client/coral-admin/bundle.js:53930:32
ObservableQuery.prototype.setUpQuery/observer.error/<@http://localhost:3000/client/coral-admin/bundle.js:24145:25
ObservableQuery.prototype.setUpQuery/observer.error@http://localhost:3000/client/coral-admin/bundle.js:24143:17
QueryManager.prototype.queryListenerForObserver/<@http://localhost:3000/client/coral-admin/bundle.js:44256:29
QueryManager.prototype.broadcastQueries/</<@http://localhost:3000/client/coral-admin/bundle.js:44719:25
QueryManager.prototype.broadcastQueries/<@http://localhost:3000/client/coral-admin/bundle.js:44716:17
QueryManager.prototype.broadcastQueries@http://localhost:3000/client/coral-admin/bundle.js:44713:9
QueryManager.prototype.broadcastNewStore@http://localhost:3000/client/coral-admin/bundle.js:44148:9
ApolloClient/this.middleware/</</<@http://localhost:3000/client/coral-admin/bundle.js:43890:25
dispatch@http://localhost:3000/client/coral-admin/bundle.js:58951:18
QueryManager.prototype.fetchRequest/retPromise</<@http://localhost:3000/client/coral-admin/bundle.js:44643:17
notify/</run@http://localhost:3000/client/coral-admin/bundle.js:27307:22
notify/<@http://localhost:3000/client/coral-admin/bundle.js:27320:28
module.exports/flush@http://localhost:3000/client/coral-admin/bundle.js:8806:9
```

File: graph/typeDefs.graphql [hunk 2]
```diff
@@ -290,6 +293,28 @@ type FlagAction implements Action {
...
 }
 
+# A DONTAGREE action that contains do not agree metadata.
+type DontAgreeAction implements Action {
+
+  # The ID of the DontAgree Action.
+  id: ID!
+
+  # The reason for which the DontAgree Action was created.
+  reason: String
```
Comment (wyattjoh) [inline graph/typeDefs.graphql:303]: Isn't the "reason" for this "I don't agree"? I don't think we need this field.

File: graph/resolvers/root_mutation.js [hunk 3]
```diff
@@ -38,6 +38,9 @@ const RootMutation = {
   createFlag(_, {flag: {item_id, item_type, reason, message}}, {mutators: {Action}}) {
     return wrapResponse('flag')(Action.create({item_id, item_type, action_type: 'FLAG', group_id: reason, metadata: {message}}));
   },
+  createDontAgree(_, {dontagree: {item_id, item_type, reason, message}}, {mutators: {Action}}) {
+    return wrapResponse('dontagee')(Action.create({item_id, item_type, action_type: 'DONTAGREE', group_id: reason, metadata: {message}}));
```
Comment (wyattjoh) [inline graph/resolvers/root_mutation.js]: `reason` shouldn't be used here, set `group_id: null`

File: graph/typeDefs.graphql [hunk 4]
```diff
@@ -534,34 +572,62 @@ input CreateFlagInput {
...
+input CreateDontAgreeInput {
+
+  # The item's id for which we are to create a don't agree.
+  item_id: ID!
+
+  # The type of the item for which we are to create the don't agree.
+  item_type: ACTION_ITEM_TYPE!
+
+  # The reason for not agreeing with the item.
+  reason: String!
```
Comment (wyattjoh) [inline graph/typeDefs.graphql]: Same as above, should remove `reason`.

**Review by wyattjoh — CHANGES_REQUESTED** [review]

File: graph/typeDefs.graphql (hunk 2 above)
Comment (davidgljay) [reply graph/typeDefs.graphql:303]: In principal we might have multiple "don't agree" reasons in the future, and in that case I wouldn't want to put a schema change. If that's not a concern then I'm happy to kill it.

**Comment by davidgljay** [comment]
@gabelula I can't reproduce the error you reported. Did you just have a don't agree comment in there?

**Review by gabelula — APPROVED** [review]

**Review by wyattjoh — APPROVED** [review]

**Pull request merged**